.github/workflows/codeql-analysis.yml

```diff
@@ -75,33 +65,49 @@ jobs:
       with:
         go-version: 1.23.x
 
+    - name: Install wasm-ld
+      run: |
+        sudo apt-get update && sudo apt-get install -y lld-14
+        sudo ln -s /usr/bin/wasm-ld-14 /usr/local/bin/wasm-ld
+
     - name: Install rust stable
       uses: dtolnay/rust-toolchain@stable
+      id: install-rust
+      with:
+        toolchain: '1.84.1'
+        targets: 'wasm32-wasip1, wasm32-unknown-unknown'
+        components: 'llvm-tools-preview, rustfmt, clippy'
+
+    - name: Install rust nightly
+      uses: dtolnay/rust-toolchain@nightly
+      id: install-rust-nightly
+      with:
+        toolchain: 'nightly-2025-02-14'
```

Comment:
Are we going to need to remember updating this from time to time?

Reply:
yup, should be updated the same time that we update stable rust version.  This version was selected to be before `wasm-wasi` was removed, has to move everything to `wasm-wasp1` before we can update.